db/icat.go

```diff
@@ -54,70 +60,143 @@ SELECT id, root FROM child_mapping WHERE storage`
 	return nil
 }
 
-func (i *ICATDatabase) createSpecificUserColls(context context.Context, username string) error {
-	u := i.UnqualifiedUsername(username)
+func (i *ICATDatabase) createUserCollsTable(context context.Context) (string, error) {
```

Comment:
Here and down there's a lot of reorganization, which I'll summarize here. I changed `createSpecificUserColls` and the newly-added `createBatchUserColls` to both call a function to create a `user_colls` temporary table and then a second one to populate it. Then, below that, the externally-callable functions call out to that as well as a few other things that I split out, so there's at least less duplication between the two code paths.